File: modules/jackson/src/main/scala/com/github/eikek/calev/jackson/CalEventDeserializer.java
package com.github.eikek.calev.jackson;

import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.FromStringDeserializer;
import com.github.eikek.calev.CalEvent;

public class CalEventDeserializer extends FromStringDeserializer<CalEvent> {
	protected CalEventDeserializer() {
		super(CalEvent.class);
	}

	@Override
	protected CalEvent _deserialize(String value, DeserializationContext ctxt) {
		return CalEvent.unsafe(value);
	}
}

File: modules/jackson/src/main/scala/com/github/eikek/calev/jackson/CalEventDeserializer.scala
package com.github.eikek.calev.jackson

import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.FromStringDeserializer
import com.github.eikek.calev.CalEvent

class CalEventDeserializer extends FromStringDeserializer[CalEvent](classOf[CalEvent]) {
  override protected def _deserialize(
      value: String,
      ctxt: DeserializationContext
  ): CalEvent =
    CalEvent.unsafe(value)
}
